File: problems/69-sqrt-of-x/csharp/69_sqrt_of_x.cs
using System.Numerics;
using System;

public static class Program 
{
	public static void Main(string[] args) 
	{
		Solution s = new Solution();
		Console.WriteLine(s.MySqrt(4));
		Console.WriteLine(s.MySqrt(8));
		Console.WriteLine(s.MySqrt(2147395599));
		Console.WriteLine(s.MySqrt(2147395600));
	}
}

public class Solution
{
    public int MySqrt(int x)
    {
        BigInteger i = new BigInteger(0);

        while (i * i <= x)
        {
            if ((i + 1) * (i + 1) > x)
                break;
            i++;
        }

        return (int)i;
    }
}

File: problems/69-sqrt-of-x/python/69_sqrt_of_x.py
class Solution:
    def mySqrt(self, x: int) -> int:
        i: int = 0

        while i ** 2 < x:
            if (i + 1) ** 2 > x:
                break
            i += 1

        return i


def test(x: int) -> int:
    r: int = Solution().mySqrt(x)
    print(f"MySqrt({x}) = {r}")


if __name__ == "__main__":
    test(4)
    test(8)
    test(2147395599)
    test(2147395600)
